app/models/post.rb

```diff
@@ -1,3 +1,14 @@
 # Represents a post belonging to an author.
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
 end
```

Comment:
- [ ] Kindly consider adding a active callback `after_save` to call the #update_posts_counter method every time a new post has been saved to database, then make the update_posts_counter method private. This means you will have to rearrange your methods so that the `recent_comments` method will be above the `update_posts_counter` so as to avoid making it private too.